test/index.js

```diff
@@ -163,38 +176,100 @@ test("htmlGenerator works with nested htmlGenerator calls in an array", () => {
 
 test("htmlGenerator works with other generators", () => {
   const generator = htmlGenerator`<div>!${generatorExample()}</div>`;
-  assert.strictEqual(generator.next().value, "<div>");
-  assert.strictEqual(generator.next().value, "<p>");
-  assert.strictEqual(generator.next().value, "This is a safe description.");
+  let accumulator = "";
+
+  for (const value of generator) {
+    accumulator += value;
+  }
+
   assert.strictEqual(
-    generator.next().value,
-    "<script>alert('This is an unsafe description.')</script>",
+    accumulator,
+    "<div><p>This is a safe description.<script>alert('This is an unsafe description.')</script>12345255</p></div>",
   );
-  assert.strictEqual(generator.next().value, "12345");
-  assert.strictEqual(generator.next().value, "255");
-  assert.strictEqual(generator.next().value, "</p>");
-  assert.strictEqual(generator.next().value, "</div>");
```

Comment:
You should have a test that verifies that X amount of chunks have been produced and not just one..

Reply:
Just added [this](https://github.com/gurgunday/ghtml/pull/12/files#diff-5bb8db779819ddef5956a5d9d5949c05ef7445237656ca37bf2f02720271440bR266-R305), simulates a file being rendered for each element in an array, each call is a readFile promise, is it good?